daemon/logs.go

```diff
@@ -18,9 +18,16 @@ import (
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
-// ContainerLogs hooks up a container's stdout and stderr streams
-// configured with the given struct.
-func (daemon *Daemon) ContainerLogs(ctx context.Context, containerName string, config *backend.ContainerLogsConfig, started chan struct{}) error {
+// ContainerLogsRawChan, given a channel, returns raw log message structs from
+// the container logger.
+func (daemon *Daemon) ContainerLogsRawChan(ctx context.Context, containerName string, config *backend.ContainerLogsConfig, msgs chan *logger.Message) error {
```

Comment:
Maybe this should be `ContainerLogs` and the other function should be moved into the API server implementation. Not sure if there are any obstacles, but it seems like the other one is more of a wrapper that doesn't need to be part of the backend.